lib/libc/minimal/source/string/strerror.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © 2005-2014 Rich Felker, et al.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* Based on: https://fossies.org/dox/musl-1.1.22/strerror_8c_source.html */
+
+#include <errno.h>
+
+#define E(a, b) ((unsigned char)a),
+static const unsigned char errid[] = {
+#include "__strerror.h"
+};
+
+#undef E
+#define E(a, b) b,
+static const char * const errmsg[] = {
+#include "__strerror.h"
+};
+
+char *strerror(int e)
+{
+	int i;
+
+	for (i = 0; errid[i] && errid[i] != e; i++) {
+		/* intentionally left blank */
+	}
+	return (char *)errmsg[i];
```

Comment:
I cannot agree about removing "const" modifier like here. It is very risky. I would prefer to have a non-standard ```strerror``` implementation that returns ```const char*``` instead of such a dirty trick.